ros2topic/ros2topic/api/__init__.py

```diff
@@ -146,12 +155,19 @@ def __call__(self, prefix, parsed_args, **kwargs):
 
 def qos_profile_from_short_keys(
     preset_profile: str, reliability: str = None, durability: str = None,
+    depth: unsigned_int = 0, history: str = None,
 ) -> rclpy.qos.QoSProfile:
     """Construct a QoSProfile given the name of a preset, and optional overrides."""
     # Build a QoS profile based on user-supplied arguments
     profile = rclpy.qos.QoSPresetProfiles.get_from_short_key(preset_profile)
+    if depth > 0:
+        profile.depth = depth
```

Comment:
I'm not so sure about this logic.  If the user passes `--qos-depth 0`, then we'll ignore their preference.  I think we'll need to use some sentinel value (maybe -1) to correspond to "the user didn't set a depth", at which point we'd use the default for the profile that they chose.

Reply:
makes sense to me, fixing.